### PR TITLE
fix: allow embedding two pages generated into the same page in "embedded" mode

### PR DESCRIPTION
**Pull request by ejmurra**
### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This PR namespaces the global object that initializes the root component in embedded mode.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

This is an attempt to fix https://github.com/sveltejs/kit/issues/9576

I wasn't sure how to write tests for this but will happily add some with some guidance.

### CODE REVIEW

**Comment by benmccann** [comment]
@ejmurra a very small note - don't put a space before the "x" when checking the boxes in the PR description. GitHub will then render them as check boxes

File: .changeset/late-toes-cry.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+'@sveltejs/kit': patch
+---
+
+fixed the generated namespace for "embedded" sveltekit apps so that two separate pages can be inlined into the same page without clobbering each other
```
Comment (benmccann) [inline .changeset/late-toes-cry.md:5]: fix messages in the changelog should start with `fix: ` for consistency: https://github.com/sveltejs/kit/blob/master/packages/kit/CHANGELOG.md

sveltekit should be capitalized as SvelteKit

**Comment by benmccann** [comment]
Can you clarify what this is fixing? Why would the timestamp not be sufficient? Or why don't you just give the apps different names? How can you be sure the two apps have different leaf nodes?

**Comment by ejmurra** [comment]
It's not two apps. It's a single app that renders two pages and tries to inline them into a separate 3rd page. Repro steps here: https://github.com/sveltejs/kit/issues/9576

I'm no svelte expert, but my understanding after spending time to create this PR is that the leaf nodes are going to be unique in the same app and that there will be one for each page. Please correct me if I'm wrong about that!

**Comment by benmccann** [comment]
Ah, sorry I missed the link to the issue. Usually people put that above the boilerplate. If you do that next time hopefully it will keep it from getting overlooked

File: .changeset/late-toes-cry.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@sveltejs/kit': patch
+---
+
+fix: namespace collision on global object when two pages generated in "embedded" mode are embedded into the same page
```
Comment (benmccann) [inline .changeset/late-toes-cry.md:5]: maybe we could shorten this a bit
```suggestion
fix: allow embedding two pages generated into the same page in "embedded" mode
```

**Review by benmccann — COMMENTED** [review]
you'll need to run `pnpm format` to fix the failing lint job

**Comment by ejmurra** [comment]
I've added an integration test for this in `test/apps/options-3/`. The test could potentially be moved into another test app but it requires config `embedded: true` and no csp directives.

**Comment by benmccann** [comment]
We try to avoid adding new test projects and just reuse existing ones. Perhaps that won't be possible here with the use of the `embedded` option though. I don't fully know if this is worth adding a test for, but if we do want to then we should make it more like a default project. It looks like you copied another test app that was testing all sorts of options and so there's a bunch of really uncommon stuff in your test project that will make things harder to understand like calling the source directory `source` instead of `src` and calling the Vite config `vite.custom.config`, etc.

**Comment by Rich-Harris** [comment]
baffled by these test failures

**Review by Rich-Harris — APPROVED** [review]

**Comment by Rich-Harris** [comment]
thank you!

**Pull request merged**